sw/device/tests/sleep_pin_wake_test.c

```diff
@@ -0,0 +1,111 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_gpio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// PLIC structures
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+static dif_rv_plic_t plic;
+
+// Volatile for vseq to assign random constant to select one of 8 MIO DIO
+static volatile const uint8_t kWakeupSel = 0;
+
+/**
+ * Pins to be configured.
+ *
+ * Assign GPIO Pads to Pinmux so that sv sequence can drive GPIO PADs.
+ */
+static const uint32_t kGpioMask = 0x000001ff;
+
+bool test_main(void) {
+  dif_pwrmgr_t pwrmgr;
+  dif_pinmux_t pinmux;
+  dif_gpio_t gpio;
+
+  dif_pinmux_index_t detector;
+  dif_pinmux_wakeup_config_t wakeup_cfg;
+
+  // Initialize power manager
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+
+  if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+    LOG_INFO("POR reset");
+
+    // TODO: Enable Pwrmgr wake up event, PLIC, then configure PINMUX randomly
```

Comment:
this lgtm!
@sriyerg recently added a random testutil that allows the C side to varied entropy from run to run also (depending on SV seed). So if you want to randomize it that way (although it might make wakeup selection a bit painful), that would work too.

Reply:
Hum.. that's interesting. @sriyerg  where can i get the examples?

Reply:
We discussed this offline.

Reply:
OK the summary is:

- include `rand_testutils` to use `rand_testutils_gen32_range(min, max)` function.
- Use LOG_INFO to write the variable (sv vseq may access to the variable via `cfg.sw_logger_vif.args[n]`.

I can try this. Thanks!